File: C_Programming/Study/Q2_Assignment31.c
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function Name :  CountSmall
// Description:     Assignment_31: Q.1 :Write a program which accept string from user and count number of Small 
//                                      Characters.
// Input :Marvellous
// Output: 9
//          
// Date:            23/06/2026
// Author:          Shital Ajit Nikam   
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<stdio.h>

int CountSmall(const char *str)
{
    int iCnt = 0;

    while(*str != '\0')
    {
        if(*str >= 'a' && *str <= 'z')
        {
            iCnt++;             
        }
        str++;        
    }

    return iCnt;

}

int main()
{
    char Arr[50] = {'\0'};
    int iRet = 0;

    printf("Enter String : \n");
    scanf("%[^'\n']s",Arr);

    iRet = CountSmall(Arr);

    printf("Frequency is : %d\n",iRet);

       
    return 0;
}

File: Python_Programming/Competative/Assignment31/Q2_Assignment31.py
#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
#        
# Description: Assignment_31: Q.2: Write a program that display message And schedule function using 
#                                   schedule.every(5).seconds.do(DisplayMessage,Message) 
# Input : 
# Message : Jay Ganesh                                
#
# Date:            21/07/2026
# Author:          Shital Ajit Nikam   
#
#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import schedule
import time
import datetime

#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
# Function Name: DisplayMessage: Schedule Message for evry 5.
#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////

def DisplayMessage(Msg):     
    
    Border = "-"*60                                    
    
    try:        
        print(Border)
        print(f"Message is :{Msg} at: {datetime.datetime.now()} \n")    # msg with current date time.
        print(Border)                    
                        
    except Exception as e :
        print("Error : ",e)                           

    
def main():
    Border = "-"*60
    print(Border) 
    print(" Marvellous Automation Script ")
    print(Border)     
    
    Message = input("Enter the Message: ")                            # Accept the Message
    
    # schedule message for specified time interval.
    schedule.every(5).seconds.do(DisplayMessage,Message)
    
    while True:        
        schedule.run_pending()                         # will check pending tasks explicitely if missed any tasks
               
        time.sleep(1)                                  # PVM will sleep for 1 sec in between.                                         
           
            
    print(Border) 
    print(" Thank you for using Marvellous Automation Script ")
    print(Border) 
        
    
if __name__ == "__main__":
    main()